### ISSUE TITLE
Ethereum upgrades to 1.8.20 sync data, data compression causes synchronous data to stop

### ISSUE THREAD

**Issue by LucasGmm** [ethereum/go-ethereum#18362]
hi，

I was having trouble syncing my data when I upgraded to version 1.8.20
The following situation occurs in the log during the Ethereum synchronization process.

```
INFO [12-24|14:14:24.470] Imported new chain segment               blocks=39 txs=5439  mgas=265.638 elapsed=8.072s    mgasps=32.905 number=5094982 hash=f4584c…a81f3a age=10mo1w4d  cache=0.00B
INFO [12-24|14:14:32.712] Imported new chain segment               blocks=40 txs=6200  mgas=304.909 elapsed=8.242s    mgasps=36.994 number=5095022 hash=dff296…d2db97 age=10mo1w4d  cache=0.00B
 WARN [12-24|14:14:41.858] Database compacting, degraded performance database=/data/eth_data/geth/chaindata
 WARN [12-24|14:15:42.133] Database compacting, degraded performance database=/data/eth_data/geth/chaindata
 WARN [12-24|14:16:42.197] Database compacting, degraded performance database=/data/eth_data/geth/chaindata
 WARN [12-24|14:17:42.788] Database compacting, degraded performance database=/data/eth_data/geth/chaindata
 WARN [12-24|14:18:43.037] Database compacting, degraded performance database=/data/eth_data/geth/chaindata
 WARN [12-24|14:19:43.461] Database compacting, degraded performance database=/data/eth_data/geth/chaindata
 WARN [12-24|14:20:43.906] Database compacting, degraded performance database=/data/eth_data/geth/chaindata
 WARN [12-24|14:21:44.201] Database compacting, degraded performance database=/data/eth_data/geth/chaindata
 WARN [12-24|14:22:44.668] Database compacting, degraded performance database=/data/eth_data/geth/chaindata
 WARN [12-24|14:23:45.280] Database compacting, degraded performance database=/data/eth_data/geth/chaindata
 ```

1. Geth version
        Geth/v1.8.20-stable-24d727b6/linux-amd64/go1.11.2
2. The 1.7.x version did not happen like this
3. If this happens, continuing to synchronize for a while after reboot will still result in data compression.
4. A similar problem on github also seems to be unresolved, seeing someone replying to version 1.8.11 after about 3.5 days of compression, starting normal synchronization 
5. I hope to get help, thanks to everyone who responded

**Comment by jordimorris** [comment]
Same issue here

**Comment by plakat3** [comment]
Well, there is no help. It is a helpless environment.

**Comment by holiman** [comment]
@plakat3 at least you're certainly not helping. Lot of people use Christmas and new year as a time to decompress and take a vacation. Sorry that doesn't meet your expectations of the software you paid nothing for

**Comment by LucasGmm** [comment]
> @ plakat3至少你当然没有帮助。很多人用圣诞节和新年作为解压缩和度假的时间。很抱歉，这不符合您对您未支付的软件的期望

@holiman Sorry, maybe my question didn't take into account the arrival of the New Year, but the current problem is more important for more Ethereum users. If there is relevant information or solution, it would be even better.

**Comment by LucasGmm** [comment]
> @plakat3 at least you're certainly not helping. Lot of people use Christmas and new year as a time to decompress and take a vacation. Sorry that doesn't meet your expectations of the software you paid nothing for


@holiman Sorry, maybe my question didn't take into account the arrival of the New Year, but the current problem is more important for more Ethereum users. If there is relevant information or solution, it would be even better.

**Comment by plakat3** [comment]
@holiman  Sorry you feel that way, but this issue has been around for a loooong time. Didn't want to spend more time researching this, but for the last year it was reported at early june, several times here, on github.
Taking point from your reply I believe that back in june a lot of people used summer as time to decompress and take a vacation. I guess after that a 'back to school' was taking place. :)
...

Long time ago I was dealing with bitcoin, litecoin, doge. With all of these I've used a software that I didn't pay for. It may surprise you, but I've never ran in any issues with these... But eth.

You may get a feeling that I'm a little upset. I'm not upset. I'm pissed off wasting my time wit a software that I didn't pay for. And as I said in another thread, if you guys don't know what you are doing, hire some people that doing this for a living, not for fun,
In other words, either do it the right way, or don't do it at all.

**Comment by mightypenguin** [comment]
Have you guys tried starting from a fresh database and resynching all of the data?
I know it's not "fun" but is it possible?

**Comment by holiman** [comment]
What syncmode is that being used? 
What is the underlying hardware - SSD or HDD ?

**Comment by plakat3** [comment]
So I'm using (trying) to run a full node with a "full" sync. Tried several times for a month, and seems like there is no difference in using either HDD or SSD. Same behavior. 
Because of a considerable difference in data transfer between these, I've replaced my HDD with 860 EVO 1TB SATA 3 SSD, No difference.
I'll not post screenshots since it happened all of the time an I dont have them, but will try to explain:
After starting a new sync (means delete all of the data) at some point geth throws an error like "hash expected -xxxxx, got 'yyyyy, and stops. that I've noticed two times. So I've deleted the last block and started from that point. Everything goes as expected until it reaches around 11m4w behind.

`INFO [01-10|06:53:13.537] Imported new block receipts              count=568  elapsed=7.521s    number=5006866 hash=bb3a6d…a56cbd age=11mo1w6d size=55.04mB
WARN [01-10|06:53:16.700] Database compacting, degraded performance database=C:\\Users\\Katy\\AppData\\Roaming\\Ethereum\\geth\\chaindata`

Apparently geth keeps downloading blocks, but up to 11 months. This is going on all the time. I'm always 11 month behind.
Geth version is 1.8.20-stable.

**Comment by holiman** [comment]
Hm, so it's obviously windows. Wonder if the db performance is degraded due to ntfs reported inability to handle many files.

**Comment by holiman** [comment]
We should have a switch to make leveldb use larger files, and windows-users can see if that works better for them. Cc @karalabe @fjl ?

**Comment by holiman** [comment]
@plakat3 just for the record, no, you're not doing a `full` sync ('syncmode=full'). That type of log output would only come through a fast-sync.

**Comment by plakat3** [comment]
> 
> 
> Maybe you didn't clean out the old data? If you already have started a sync, it won't restart with another type of sync. If you're up for it, consider giving #18203 a spin and see if it works better for you

Thanks, Will take a look and will report 
:)

**Comment by holiman** [comment]
If you do, use it on top of latest master, and not the actual PR which is getting old. It's a trivial change, should be easy to add.

**Comment by plakat3** [comment]
> 
> 
> EDIT: and it probably makes sense to do a fast-sync, actually. A true full-sync takes quite some time more than fast-sync does, takes more space, and does not offer any advantages really for a miner.

For sure I've cleaned everything each time I've restarted this. It gets funny at some point because as of the beginning, did some research and most of the articles suggest that I should run a full mode to be able to mine.  That was the old fashion way so I assumed fairly that this is it.

**Comment by holiman** [comment]
@rjl493456442 is investigating compaction issues and doing various improvements to leveldb. See e,g, https://github.com/ethereum/go-ethereum/pull/19201 ,

**Comment by holiman** [comment]
As we've done quite extensive work on db compaction for the 1.9 branch (soon to be released, and has been on master for a long time), I'll consider this closed. Please reopen if you're still seeing this problem on master
